### DISCUSSION TITLE
Uploading noise/static plus   dfmp3.reset(); hangs

### DISCUSSION THREAD

**Discussion by 26tajeen** [Makuna/DFMiniMp3#147] [Q&A]
This might be something I've done stupidly.

When I upload to the board I can hear a LOAD of noise (sounds like a dialup modem!) until it's done.

if this line   dfmp3.reset(); is present in the code then the code never gets beyond 'initializing'.

If I remove it I can get my sketch to run as normal, but 0 files are reported over serial and the volume set doesn't work. Volume is always reported as 0.

Any ideas?!

Thank you!

**Comment by Makuna** [comment]
Unfortunately, these modules are little black boxes with little documentation or best practices, that is probably why you are here ;-)

Not sure about upload as the modules I have don't have flash memory present/supported on them.  It might be normal.  Do you hear the sound even if you mute it during the upload?  See disableDac()/enableDac().  You might also try changing the playback source during upload, see setPlayBackSource().

Calling reset in final product isn't suggested. [See Wiki Here](https://github.com/Makuna/DFMiniMp3/wiki/API-Reference#void-reset)

You may try calling getStatus() and see what state the module is in when you don't call reset, it may shed some light on why volume is not working.
Also see the wiki for the sleep() call, where calling setPlaybackSource(DfMp3_PlaySource_*) will wake the module up; this may also do some minor reset internal.
